src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/PackageReferenceArgs.cs

```diff
@@ -14,34 +11,28 @@ public class PackageReferenceArgs
         public string ProjectPath { get; }
         public ILogger Logger { get; }
         public bool NoVersion { get; set; }
-        public PackageDependency PackageDependency { get; set; }
         public string DgFilePath { get; set; }
         public string[] Frameworks { get; set; }
         public string[] Sources { get; set; }
         public string PackageDirectory { get; set; }
         public bool NoRestore { get; set; }
-
         public bool Interactive { get; set; }
-
-        public PackageReferenceArgs(string projectPath, PackageDependency packageDependency, ILogger logger, bool noVersion)
+        public bool Prerelease { get; set;}
+        public string PackageId { get; set; }
+        public string PackageVersion { get; set; }
+        
+        public PackageReferenceArgs(string projectPath, ILogger logger, bool noVersion)
```

Comment:
Consider overloading rather than eliminating the old constructor, as this is count as as shipped API.

If you think this constructor is better, mark the other ones as obsolete

Reply:
NuGet.CommandLine.Xplat is an exe.
No public APIs.

Reply:
No need, NuGet.CommandLine.Xplat is treated as an executable, https://github.com/NuGet/NuGet.Client/blob/28f5e5c139daf85a4e3530a0bbd1f8c2cc1f9c94/src/NuGet.Core/NuGet.CommandLine.XPlat/NuGet.CommandLine.XPlat.csproj#L14. 

cc @zivkan would know better.

Reply:
`NuGet.CommandLine.XPlat` is designed to be an `exe`, not a class library that other assemblies call directly. However, we do publish it as a dependency package that can be referenced by other projects (and I did find one case on github where someone is doing that, and my memory is there were comments from customers in NuGet/Home where others say the same).

I'm really in two minds, I can't decide.

Ideally I would search all `public` modifiers and change them to `internal`, so that there's nothing public in the assembly at all. In that case changing the constructor wouldn't matter because it's not an API breaking change. But we already shipped packages with public APIs, so this would be a breaking change.

But we live in reality. Customers might be depending on the old constructor, we have no way of knowing. In my opinion, there's nothing that makes it obvious to customers that they should not be using this package in their own projects. So, if people are, maybe we should try to "do the right thing" and maintain compatibility.

A third option is that we just pretend that `NuGet.CommandLine.XPlat` is not shipped as a package, we break APIs whenever we want, and if a customer complains, we try our best to explain. The idealist in me doesn't like this. I'd lean more towards making everything internal, and breaking everyone once-off.

Related discussion here: https://github.com/NuGet/NuGet.Client/pull/3512#pullrequestreview-449198867

Reply:
Adding my comment from the other thread. 

I see this as an equivalent of NuGet.exe. So whatever approach allows to avoid comments like the above that'd make us preserve outdated constructors.

Reply:
This is what I've decided, unless someone screams at me:  This PR can break the constructor. After it's merged I'll create a PR to change everything public to internal. When my PublicAPIAnalyzer PR gets merged later, then it will kind of "enforce" that this XPlat library should keep everything internal (plus I'll mention it explicitly in the docs)